media/js/teach/smarton.js

```diff
@@ -0,0 +1,303 @@
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof window.Mozilla === 'undefined') {
+    window.Mozilla = {};
+}
+
+;(function($) {
+    'use strict';
+
+    var $window = $(window);
+    var $document = $(document);
+    var $body = $('body');
+    var $header = $('.page-header');
+    var $nav = $('.nav-steps');
+    var $firstSection = $('#ask');
+    var $firstStep = $('#nav-step-ask');
+    if ($header.length > 0) {
+        var headTop = $header.offset();
+        var headHeight = $header.height();
+    }
+
+    if (window.isFirefox()) {
+        $body.addClass('is-firefox');
+    } else {
+        $body.addClass('not-firefox');
+    }
+
+    // Sticky navigation
+    var fixed = false;
+    var didScroll = false;
+
+    $window.scroll(function() {
+        didScroll = true;
+    });
+
+    if ($header.length > 0) {
+        $(document).ready(function() {
+            var scrollTop = $window.scrollTop();
+            if (scrollTop >= headTop.top) {
+                didScroll = true;
+            }
+        });
+    }
+
+    function adjustScrollbar() {
+        if (didScroll) {
+            didScroll = false;
+            var scrollTop = $window.scrollTop();
+            if ($header.length > 0) {
+                if( scrollTop >= headTop.top ) {
+                    if(!fixed) {
+                        fixed = true;
+                        $header.addClass('fixed');
+                        $firstStep.addClass('current');
+                        // recalculate height after sticking
+                        headHeight = $header.height();
+                        // then pad the first section so it doesn't jump as far
+                        $firstSection.css({ paddingTop: headHeight + 40 });
+                    }
+                } else {
+                    if(fixed) {
+                        fixed = false;
+                        $header.removeClass('fixed');
+                        $firstStep.removeClass('current');
+                        $firstSection.removeAttr('style');
+                        $header.find('.nav-steps li').removeClass();
+                    }
+                }
+            }
+        }
+    }
+
+    // Check for an adjusted scrollbar every 100ms.
+    setInterval(adjustScrollbar, 100);
+
+    // Scroll smoothly to the linked section
+    $document.on('click', '.nav-steps a[href^="#"]', function(e) {
+        e.preventDefault();
+        // Extract the target element's ID from the link's href.
+        var elem = $(this).attr('href').replace( /.*?(#.*)/g, "$1" );
+        $('html, body').animate({
+            scrollTop: $(elem).offset().top - 83
+        }, 1000, function() {
+            $(elem).attr('tabindex','100').focus().removeAttr('tabindex');
+        });
+    });
+
+    var $footer = $('#footer-cta');
+    var footHeight = $footer.height();
+    var tallMode = false;
+    var queryHeight = matchMedia('(min-height: 650px)');
```

Comment:
This throws an error is all browsers that don't support `matchMedia`. Ideally we should move all matchMedia logic inside it's own function, and only call it after using a feature detect.